File: Transport list (C++)/Transport.cpp
#include "Transport.h"


Transport::Transport(Model model, Engine engine, Transmission transmission, Body body, int speed, int passengers, int fuel)
{
	this->model = model;
	this->engine = engine;
	this->transmission = transmission;
	this->body = body;
	this->maxSpeed = speed;
	this->passengersNumber = passengers;
	this->fuelLevel = fuel;
}

void Transport::Print()
{
	this->model.PrintModel();
	this->engine.PrintEngine();
	this->transmission.PrintTransmission();
	this->body.PrintBody();
	cout << "Максимальная скорость:\t" << this->GetMaxSpeed() << "\n";
	cout << "Пассажировместимость:\t" << this->GetPassengers() << "\n";
	cout << "Объем бака:\t" << this->GetFuel() << "\n";

}

///////////////////////////////////////////////

void Transport::SetEngine(Engine other)
{
	this->engine = other;
}

void  Transport::SetTransmission(Transmission other)
{
	this->transmission = other;
}

void  Transport::SetModel(Model other)
{
	this->model = other;
}

void  Transport::SetBody(Body other)
{
	this->body = other;
}


void Transport::SetMaxSpeed(int speed)
{
	this->maxSpeed = speed;
}

void Transport::SetPassengers(int passengers)
{
	this->passengersNumber = passengers;
}

void Transport::SetFuel(float fuel)
{
	this->fuelLevel = fuel;
}

///////////////////////////////////////////////

Engine Transport::GetEngine()
{
	return this->engine;
}

Model Transport::GetModel()
{
	return this->model;
}

Transmission Transport::GetTransmission()
{
	return this->transmission;
}

Body Transport::GetBody()
{
	return this->body;
}

int Transport::GetMaxSpeed()
{
	return this->maxSpeed;
}

int Transport::GetPassengers()
{
	return this->passengersNumber;
}

float Transport::GetFuel()
{
	return this->fuelLevel;
}

///////////////////////////////////////////////////////

string Transport::WhatTransport()
{
	return "\n++++ Транспорт перемещается ++++\n";
}

Transport& Transport::operator=(const Transport& other)
{
	if (this != &other)
	{
		this->fuelLevel = other.fuelLevel;
		this->passengersNumber = other.passengersNumber;
		this->maxSpeed = other.maxSpeed;
	}

	return *this;
}

File: Transport list (Java)/classes/Transport.java
package classes;

// Класс "Транспорт"
public abstract class Transport
{
    protected int speed; // Скорость
	protected int passengersNumber; // Количество пассажиров
	protected float fuelLevel; // Уровень топлива
    protected int id;
    protected Model model;
    protected Engine engine;
    protected Transmission transmission;
    protected Body body;

	public Transport() {};
	public Transport(Model model, Engine engine, Transmission transmission, Body body, int speed, int passengers, float fuel)
    {
        this.model = model;
        this.engine = engine;
        this.transmission = transmission;
        this.body = body;
        this.speed = speed;
        this.passengersNumber = passengers;
        this.fuelLevel = fuel;
    }

    public Transport(Transport transport)
    {
        this.model = transport.model;
        this.engine = transport.engine;
        this.transmission = transport.transmission;
        this.body = transport.body;
        this.speed = transport.speed;
        this.passengersNumber = transport.passengersNumber;
        this.fuelLevel = transport.fuelLevel;
    }
	
	// Получить максимальное количество пассажиров
	public int GetMaxPassengers()
	{ 
		return this.passengersNumber; 
	};

    public int GetMaxSpeed()
    {
        return this.speed;
    };

    // Получить максимальный уровень топлива
	public float GetMaxFuelLevel() 
    {
         return this.fuelLevel;
    }; 

    // Получение двигателя (Нужно для прикладной функции)
    public Engine GetEngine()
    {
        return this.engine;
    }

    // Получение коробки передач (Нужно для прикладной функции)
    public Transmission GetTransmission()
    {
        return this.transmission;
    }

    // Получение модели (Нужно для прикладной функции)
    public Model GetModel()
    {
        return this.model;
    }

    // Получение кузова (нужно, чтобы через класс body получить количество колес)
    public Body GetBody()
    {
        return this.body;
    }

    public void Print()
    {
        System.out.print(this.model.toString());
        System.out.print(this.engine.toString());
        System.out.print(this.transmission.toString());
        System.out.print(this.body.toString());
        System.out.printf("\nМаксимальная скорость:\t%d", this.speed);
        System.out.printf("\nПассажировместимость:\t%d", this.passengersNumber);
        System.out.printf("\nОбъем бака:\t%.2f", this.fuelLevel);    
    }
};